packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -290,7 +290,10 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
     window.onLocaleChanged = handleLocaleChanged;
     window.onAccessibilityFeaturesChanged = handleAccessibilityFeaturesChanged;
     SystemChannels.navigation.setMethodCallHandler(_handleNavigationInvocation);
-    FlutterErrorDetails.propertiesTransformers.add(transformDebugCreator);
+    assert(() {
+      FlutterErrorDetails.propertiesTransformers.add(transformDebugCreator);
```

Comment:
Maybe we should rename `transformDebugCreator` to `debugTransformDebugCreator` to make it clear that it should only be used in debug builds. Additionally, the implementation of the method could be wrapped in an assert as well so that even if you accidentally call this method in non-debug code, it doesn't actually do anything or retain code...

Reply:
Done.

I've added the test harness, which unfortunately required reving pub packages.

I can split this into a smaller PR if need be.